File: store-java/src/main/java/io/mb/store/controller/ItemsController.java
package io.mb.store.controller;

import io.mb.store.resource.dto.ItemCreateDto;
import io.mb.store.resource.dto.ItemDto;
import io.mb.store.service.StoreService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.*;

import java.util.List;

import static io.mb.store.resource.Constants.ITEMS_MAPPING;

@RestController
@RequestMapping(ITEMS_MAPPING)
@RequiredArgsConstructor
@Slf4j
@Validated
public class ItemsController {
    private final StoreService storeService;

    @GetMapping
    public List<ItemDto> findAll() {
        log.info("Fetching all items in store");
        final List<ItemDto> itemDtos = storeService.findAll();
        log.info("Returned all items in store");
        return itemDtos;
    }

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public ItemDto create(@Valid @RequestBody final ItemCreateDto itemCreateDto) {
        log.info("Adding entry " + itemCreateDto + " to the store");
        ItemDto itemDto = storeService.save(itemCreateDto);
        log.info("Entry " + itemDto + " created and added to the store");
        return itemDto;
    }
}

File: store-kotlin/src/main/kotlin/io/mb/store/controller/ItemsController.kt
package io.mb.store.controller

import io.mb.store.resource.ITEMS_MAPPING
import io.mb.store.resource.dto.ItemCreateDto
import io.mb.store.resource.dto.ItemDto
import io.mb.store.service.StoreService
import jakarta.validation.Valid
import mu.KotlinLogging
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping(ITEMS_MAPPING)
class ItemsController(
    private val storeService: StoreService,
) {

    private companion object {
        private val logger = KotlinLogging.logger { }
    }

    @GetMapping
    fun findAll(): List<ItemDto> {
        logger.info { "Fetching all items in store" }
        return storeService.findAll().also {
            logger.info { "Returned all items in store" }
        }
    }

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    fun create(@Valid @RequestBody itemCreateDto: ItemCreateDto): ItemDto {
        logger.info { "Adding entry $itemCreateDto to the store" }
        return storeService.save(itemCreateDto).also {
            logger.info { "Entry $it created and added to the store" }
        }
    }
}